pkg/providers/snow/reconciler/reconciler.go

```diff
@@ -2,19 +2,38 @@ package reconciler
 
 import (
 	"context"
+	"fmt"
 
 	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 
 	anywherev1 "github.com/aws/eks-anywhere/pkg/api/v1alpha1"
 	"github.com/aws/eks-anywhere/pkg/controller"
 )
 
-type Reconciler struct{}
+type Reconciler struct {
+	client client.Client
+}
 
-func New() *Reconciler {
-	return &Reconciler{}
+func New(client client.Client) *Reconciler {
+	return &Reconciler{
+		client: client,
+	}
 }
 
-func (v *Reconciler) Reconcile(ctx context.Context, log logr.Logger, cluster *anywherev1.Cluster) (controller.Result, error) {
+func (r *Reconciler) Reconcile(ctx context.Context, log logr.Logger, cluster *anywherev1.Cluster) (controller.Result, error) {
+	for _, ref := range cluster.MachineConfigRefs() {
```

Comment:
we could just use `cluster.BuildSpec` here, which should already have all the snow config machines
and it will come in handy for the rest of the reconciliation loop